jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -254,6 +240,56 @@ static JibContainerBuilder processCommonConfiguration(
     return jibContainerBuilder;
   }
 
+  /**
+   * Returns a {@link JavaContainerBuilder} with the correctly parsed base image configuration.
+   *
+   * @param rawConfiguration contains the base image configuration
+   * @param projectProperties used for providing additional information
+   * @param inferredAuthProvider provides inferred auths for registry images
+   * @return a new {@link JavaContainerBuilder} with the configured base image
+   * @throws IncompatibleBaseImageJavaVersionException when the Java version in the base image is
+   *     incompatible with the Java version of the application to be containerized
+   * @throws InvalidImageReferenceException if the base image configuration can't be parsed
+   * @throws FileNotFoundException if a credential helper can't be found
+   */
+  @VisibleForTesting
+  static JavaContainerBuilder getJavaContainerBuilderWithBaseImage(
+      RawConfiguration rawConfiguration,
+      ProjectProperties projectProperties,
+      InferredAuthProvider inferredAuthProvider)
+      throws IncompatibleBaseImageJavaVersionException, InvalidImageReferenceException,
+          FileNotFoundException {
+    // Use image configuration as-is if it's a local base image
+    String baseImageConfig =
+        rawConfiguration.getFromImage().orElse(getDefaultBaseImage(projectProperties));
+    if (baseImageConfig.startsWith(Jib.DOCKER_DAEMON_IMAGE_PREFIX)
```

Comment:
I wonder if it may be OK to check Java versions for Docker images. What's the rationale to exclude here?

Reply:
Yeah, I guess no reason not to.